runtime/trappist/src/constants.rs

```diff
@@ -58,8 +60,38 @@ pub mod fee {
 	/// Yet, it can be used for any other sort of change to weight-fee. Some examples being:
 	///   - Setting it to `0` will essentially disable the weight fee.
 	///   - Setting it to `1` will cause the literal `#[weight = x]` values to be charged.
+	///
+	/// TODO: Once the runtime is upgraded to polkadot v0.9.42 or above refactor this using
+	/// the FeePolynomial struct that already includes the methods to make this calculations
+	/// and remove the custom WeightCoefficientCalc inside ./trappist/src/impls.rs
 	pub struct WeightToFee;
-	impl WeightToFeePolynomial for WeightToFee {
+	impl frame_support::weights::WeightToFee for WeightToFee {
+		type Balance = Balance;
+
+		fn weight_to_fee(weight: &Weight) -> Self::Balance {
+			let ref_poly: smallvec::SmallVec<[WeightToFeeCoefficient<Balance>; 4]> =
+				RefTimeToFee::polynomial();
+			let proof_poly: smallvec::SmallVec<[WeightToFeeCoefficient<Balance>; 4]> =
+				ProofSizeToFee::polynomial();
+
+			// Get fee amount from ref_time based on the RefTime polynomial
+			let ref_fee: Balance = ref_poly.iter().fold(0, |acc, term| {
+				term.saturating_eval(acc, Balance::saturated_from(weight.ref_time()))
```

Comment:
`Balance::saturated_from(weight.ref_time()` can be moved out of the closure, so you don't execute the same  cast on every coefficient.

Reply:
You are right, this is directly taken from [Substrate - FeePolynomial](https://paritytech.github.io/substrate/master/src/sp_weights/lib.rs.html#184) adapting it to this version, but this should also be changed in substrate.

Good catch!